hastory-server-data/src/Hastory/Server/Data/Username.hs

```diff
@@ -34,9 +35,12 @@ instance ToJSON Username where
   toJSON = toJSON . rawUserName
 
 instance Validity Username where
-  validate userName = check notNull "Username cannot be null"
+  validate userName =
+    mconcat
+      [check notNull "Username cannot be null", check allAlphaNum "Username must be alphanumeric"]
     where
       notNull = not . T.null . rawUserName $ userName
+      allAlphaNum = T.all isAlphaNum (rawUserName userName)
```

Comment:
This doesn't help with telling you which character is not `isAlphaNum`. Try using `decorateList (T.unpack t)`